market_maker_keeper/dydx_market_maker_keeper.py

```diff
@@ -112,10 +129,64 @@ def place_order_function(new_order_to_be_placed):
+                         new_order_to_be_placed.price, amount)
 
         for new_order in new_orders:
-            self.order_book_manager.place_order(lambda new_order=new_order: place_order_function(new_order))
+            if self._should_place_order(new_order):
+                self.order_book_manager.place_order(lambda new_order=new_order: place_order_function(new_order))
+            else:
+                logging.info(f"New Order below size minimum. Order ignored.")
+
+    def synchronize_orders(self):
+        bands = Bands.read(self.bands_config, self.spread_feed, self.control_feed, self.history)
+        order_book = self.order_book_manager.get_order_book()
+        target_price = self.price_feed.get_price()
+
+        # Cancel orders
+        cancellable_orders = bands.cancellable_orders(our_buy_orders=self.our_buy_orders(order_book.orders),
+                                                      our_sell_orders=self.our_sell_orders(order_book.orders),
+                                                      target_price=target_price)
+        if len(cancellable_orders) > 0:
+            self.order_book_manager.cancel_orders(cancellable_orders)
+            return
+
+        # Do not place new orders if order book state is not confirmed
+        if order_book.orders_being_placed or order_book.orders_being_cancelled:
+            self.logger.debug("Order book is in progress, not placing new orders")
+            return
+
+        our_buy_balance = self.our_available_balance(order_book.balances, self.token_buy())
+        our_sell_balance = self.our_available_balance(order_book.balances, self.token_sell())
+        total_in_buy_orders = total_buy_amount(self.our_buy_orders(order_book.orders))
+        total_in_sell_orders = total_sell_amount(self.our_sell_orders(order_book.orders))
+        our_buy_orders = self.our_buy_orders(order_book.orders)
+        our_sell_orders = self.our_sell_orders(order_book.orders)
+
+        # check that placing buy new orders won't require margin
+        for band in bands.buy_bands:
+            orders = [order for order in our_buy_orders if band.includes(order, target_price.buy_price)]
+            total_amount = total_buy_amount(orders)
+            if total_amount < band.min_amount:
+                pay_amount = Wad.min(band.avg_amount - total_amount, our_buy_balance)
+                if total_in_buy_orders + pay_amount > our_buy_balance:
+                    band.min_amount = Wad(0)
+
+        # check that placing new sell orders won't require margin
+        for band in bands.sell_bands:
+            orders = [order for order in our_sell_orders if band.includes(order, target_price.sell_price)]
+            total_amount = total_sell_amount(orders)
+            if total_amount < band.min_amount:
+                pay_amount = Wad.min(band.avg_amount - total_amount, our_sell_balance)
+                if total_in_sell_orders + pay_amount > our_sell_balance:
+                    band.min_amount = Wad(0)
```

Comment:
Setting `band.min_amount` to 0 will prevent a new order from being created on that iteration: https://github.com/makerdao/market-maker-keeper/blob/master/market_maker_keeper/band.py#L359